components/org.wso2.carbon.security.caas/src/main/java/org/wso2/carbon/security/caas/user/core/bean/User.java

```diff
@@ -33,20 +33,45 @@
  */
 public class User {
 
+    /**
+     * The globally unique userId of the user.
+     */
     private String userId;
+
+    /**
+     * The domain this user belongs to.
+     */
     private Domain domain;
-    private String tenantDomain;
 
+    /**
+     * The primary claim attribute value of this user.
+     *
+     * The primary attribute value will be most frequent claim to be retrieved from a user.
+     * Hence keeping is as an attribute.
+     */
+    private String primaryAttributeValue;
```

Comment:
It cannot be changed since the builder pattern is used to create User objects and they are immutable.